Add support for Eurofighter and Rafale mod aircraft

angelatthetomb opened the pull request:
Edited db.py, faction.py, start_generator.py, ai_flight_generator.py, qmodsettings.py, qnewgamegenerator.py. Added eurofighter.py and rafale.py. Added squadrons and custom loadouts. Tested in my forked build and work in-game.

angelatthetomb committed:
Add support for Eurogamer and Rafale mod aircraft

Khopa commented:
Hello, 

Thanks, the code changes looks good :)

However, i'm unsure which mod version is being used, can you confirm if it's these one : 

- Rafale : https://forums.eagle.ru/topic/252774-rafales-pack-new-version-005-for-279-by-the-cuesta-brothers/
- EuroFighter : https://forums.eagle.ru/topic/290086-new-version-eurofighter-typhoon-by-lechuzas-negras-v105/

Khopa commented:
💹 I've been able to test it succesfully with the Typhoon mod.

Just two things : 
* UI icons are missing
* There are no default factions allowing the use of the mod.

![image](https://user-images.githubusercontent.com/2546901/153500279-9320b09b-8dca-48dd-a6df-b600b30ac70e.png)

Could you add some for the modded planes ? (in resources/ui/aircrafts/icons)
Should be named {aircraft_class_id}_24.jpg and be a 91x24 jpeg.

Example for Typhoon: 
![Eurofighter_24](https://user-images.githubusercontent.com/2546901/153500126-5310eca5-56aa-4e40-9c59-88c441f4aeef.jpg)

(My process to produce these icons is just to screenshoot them in game, and resize in an image editor)

***

- If you want me to add the icons and default factions, i'll merge your PR and add them.
- However, if you want to do it, you can push the additional icons and factions files to this PR

angelatthetomb commented:
Hey Khopa,

I'll get some icons and a faction added to the pull request here in the next day or two. Thanks!

angelatthetomb committed:
Add support for Luchas Eurofighter, Cuesta Bros Rafale, VSN EA-6B and VSN FA-18F

Hornet2041 commented:
Um, your commits also say they add the VSN Prowler and F/A-18F.

Hornet2041 commented:
> 

![image](https://user-images.githubusercontent.com/55866858/153730500-d9058316-f22b-49f0-ad70-f9c197ba7a8d.png)
Downloaded your latest build and got that error on trying to launch a new game.

RndName commented on game/db.py:
```diff
@@ -43,18 +43,28 @@
+plane_map["EA-6B"] = EA_6B
+plane_map["Eurofighter"] = Eurofighter
+plane_map["FA18F_hornet"] = VSN_FA18F
+plane_map["FA18F_hornetAG"] = VSN_FA18F_AG
```
Neither EA-6B nor VSN_FA18F / AG are mentioned in this PR.

RndName commented on game/factions/faction.py:
```diff
@@ -275,6 +277,9 @@ def apply_mod_settings(self, mod_settings: ModSettings) -> Faction:
+        if not mod_settings.rafale:
+            self.remove_aircraft("Rafale_B")
+            self.remove_aircraft("Rafale_C")
         if not mod_settings.su57_felon:
```
Forgot to handle the EA-6B and FA-18F.

RndName reviewed (changes requested):
As already mentionend by @Hornet2041 there are unrelated mods coming in as well which are not mentionend in this PR. Please clean this one up.
Also it's kinda broken as the EA-6B and FA-18F will not be removed from the plane map when the mod is not checked.

RndName commented on resources/factions/bluefor_modern_2015.json:
```diff
@@ -0,0 +1,142 @@
+{
+  "country": "Combined Joint Task Forces Blue",
+  "name": "Bluefor Modern 2015",
```
Why a new faction? And this one is formatted quite wrong. A lot of indents are not right.

angelatthetomb committed:
Add support for Luchas Eurofighter and Cuesta Bros Rafale

angelatthetomb committed:
Merge remote-tracking branch 'origin/pulltomaster' into pulltomaster

angelatthetomb committed:
Revert "Merge remote-tracking branch 'origin/pulltomaster' into pulltomaster"

This reverts commit 1ff12c47a1b506c675b314f578d5b8108334f223, reversing
changes made to d142fa47cf671d86d9dda37a31869dc6a0f18f79.

angelatthetomb commented:
Hey guys, sorry about the EA6B and FA18 references, I committed to the wrong fork. Those are in my next project.

For now, I've got this PR back to the correct commit, and from there I added icons for the Rafale B and C and the Eurofighter. I also added them to the "Bluefor Modern" faction. Forgive me, I'm just getting to know Github, never used it before. Thanks!

DanAlbert commented on game/db.py:
```diff
@@ -43,18 +43,23 @@
 from pydcs_extensions.su57.su57 import Su_57
 
 plane_map["A-4E-C"] = A_4E_C
+plane_map["Eurofighter"] = Eurofighter
```
This ugly system was just redone. Instead, decorate mod classes with `@planemod` and be sure to add an `import` in `pydcs_extensions.__init__`. Have a look at other mods for examples.

RndName commented:
PR got stale, any further changes planned or shall we close this one?

DanAlbert commented:
We're closing stale PRs. If this is still needed, please update and reopen.
